esmvaltool/diag_scripts/lifetime/lifetime_base.py

```diff
@@ -0,0 +1,875 @@
+                else:
+                    cubelist[i] = (((lev - plev[i + 1]) / 2.
+                                   + (plev[i - 1] - lev) / 2.)
+                                   * cubelist[i] / cubelist[i])
+            cubelist[i].units = 'Pa'
+            cubelist[i].var_name = 'air_pressure'
+            cubelist[i].add_aux_coord(iris.coords.AuxCoord(lev))
+        # merge to single cube
+        dplev = iris.cube.CubeList(cubelist).merge_cube()
+
+    else:
+        raise NotImplementedError("Function not implemented"
+                                  f" for type {type(pmax)}")
+
+    return dplev
+
+
+def calculate_lifetime(dataset, plot_type, region):
+    """Calculate the lifetime for the given plot_type and region."""
+    # extract region from weights and reaction
+    reaction = extract_region(dataset, region, case='reaction')
+    weight = extract_region(dataset, region, case='weight')
+
+    # calculate nominator and denominator
+    # and sum of nominator and denominator via plot_type dimensions
+    nominator = sum_up_to_plot_dimensions(weight, plot_type)
+    denominator = sum_up_to_plot_dimensions(weight * reaction, plot_type)
+
+    # division
+    division = nominator / denominator
+
+    return division
+
+
+def extract_region(dataset, region, case='reaction'):
+    """Return cube with everything outside region set to zero.
+
+    If z_coord is defined as:
+    - air_pressure, use:
+                      - ptp and air_pressure
+                      - tp_clim and air_pressure
+    - atmosphere_hybrid_sigma_pressure_coordinate, use:
+                      - tp_i and model_level_number
+                      - ptp and (derived) air_pressure
+                      - tp_clim and (derived) air_pressure
+
+    Current aware regions:
+    - TROP: troposphere (excl. tropopause), requires tropopause pressure
+    - STRA: stratosphere (incl. tropopause), requires tropopause pressure
+    """
+    var = dataset[case]
+    z_coord = dataset['z_coord']
+
+    if (
+            'ptp' not in dataset['variables']
+            and 'tp_i' not in dataset['variables']):
+        tp_clim = climatological_tropopause(var[:, 0, :, :])
+
+    if region in ['TROP', 'STRA']:
+        use_z_coord = 'air_pressure'
+        if z_coord.name() == 'air_pressure':
+            if 'ptp' in dataset['variables']:
+                tropopause = dataset['variables']['ptp']
+            else:
+                tropopause = tp_clim
+        elif z_coord.name() == 'atmosphere_hybrid_sigma_pressure_coordinate':
+            if 'tp_i' in dataset['variables']:
+                tropopause = dataset['variables']['tp_i']
+                use_z_coord = 'model_level_number'
+            elif 'ptp' in dataset['variables']:
+                tropopause = dataset['variables']['ptp']
+            else:
+                tropopause = tp_clim
+
+        z_4d = broadcast_to_shape(
+            var.coord(use_z_coord).points,
```

Comment:
```suggestion
            var.coord(use_z_coord).lazy_points(),
```

This will always give a dask array